packages/ember-metal/lib/property_get.js

```diff
@@ -76,7 +76,7 @@ export function _getPath(root, path) {
 
   for (let i = 0; i < parts.length; i++) {
     if (obj == null) {
```

Comment:
To be specific, I would check falsy, and that the typeof was 'string', 'object', 'function' but at a minimum this can simply check falsy.  I'm sure the `obj == null` check was a bad fix to this sometimes returning false vs undefined.